runtime/vm/classsupport.c

```diff
@@ -378,10 +378,42 @@ internalFindClassString(J9VMThread* currentThread, j9object_t moduleName, j9obje
 
 			result = internalFindClassInModule(currentThread, j9module, utf8Name, utf8Length, classLoader, options);
 		}
+
+		for (int i = 0; i < utf8Length; i++) {
```

Comment:
There is no need to go through the for loop if the result is null or it is an array. The for loop can be moved after the null check and array check.

Reply:
You don't need this for loop if `J9_STR_XLAT` is already set in `stringFlags`.